regulations/views/preamble.py

```diff
@@ -0,0 +1,55 @@
+from django.http import Http404
+from django.template.response import TemplateResponse
+from django.views.generic.base import View
+
+from regulations.generator.api_reader import ApiReader
+from regulations.generator.generator import LayerCreator
+from regulations.generator.html_builder import HTMLBuilder
+
+
+def find_subtree(root, label_parts):
+    """Given a nested tree and a label to look for, find the associated node
+    in the tree. Note that, unlike regulations, preamble trees _always_ encode
+    their exact position in the tree."""
+    cursor = root
+    while cursor and cursor['label'] != label_parts:
+        next_cursor = None
+        for child in cursor['children']:
+            if child['label'] == label_parts[:len(child['label'])]:
+                next_cursor = child
```

Comment:
You're right that labels should be unique -- we'd actually get a consistency error when writing the preamble if this were not the case. We could `break` early, but meh. Unless it's confusing, I don't think the execute cycles (to finish off the child list) is worth the inherently evil `break`.

Correct, if we don't find the label, this implicitly returns None. Seems like another area for a 404.